src/Renderer/ThreeExtended/Feature2Mesh.js

```diff
@@ -121,41 +120,49 @@ function addFaces(indices, length, offset) {
 }
 
 function coordinateToPoints(coordinates, properties, options) {
-    let vertices = [];
-    let colors = [];
+    const vertices = new Float32Array(3 * coordinates.coordinates.length);
+    const colors = new Uint8Array(3 * coordinates.coordinates.length);
     const geometry = new THREE.BufferGeometry();
+    let offset = 0;
 
     /* eslint-disable guard-for-in */
     for (const id in coordinates.featureVertices) {
         const { contour, property } = extractFeature(coordinates, properties, id);
         // get altitude from properties
         const altitude = getAltitude(options, property);
         const newVertices = coordinatesToVertices(contour, altitude);
-        vertices = vertices.concat(newVertices);
+        for (let i = 0; i < newVertices.length; i++) {
```

Comment:
You should instead change `coordinatesToVertices` signature and add an `target` optionnal parameter.

Then use it like this: 
```js
coordinatesToVertices(contour, altitude, new Float32Array(vertices.buffer, offset * 4));
```

Reply:
(Or pass the target and offset as arguments if you prefer)

Reply:
thanks for your suggestion, I've updated the code !